nck/readers/mytarget_reader.py

```diff
@@ -0,0 +1,408 @@
+# GNU Lesser General Public License v3.0 only
+# Copyright (C) 2020 Artefact
+# licence-information@artefact.com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+import click
+import requests
+from typing import Dict, List, Any, Tuple
+import logging
+import itertools
+import time
+from datetime import datetime
+from nck.commands.command import processor
+from nck.readers.reader import Reader
+from nck.streams.json_stream import JSONStream
+from nck.utils.args import extract_args
+from nck.helpers.mytarget_helper import REQUEST_CONFIG
+
+
+@click.command(name="read_mytarget")
+@click.option("--mytarget-client-id", required=True)
+@click.option("--mytarget-client-secret", required=True)
+@click.option("--mytarget-mail", required=True)
+@click.option("--mytarget-agency", required=True)
+@click.option("--mytarget-refresh-token", required=True)
+@click.option("--mytarget-start-date", type=click.DateTime(), required=True)
+@click.option("--mytarget-end-date", type=click.DateTime(), required=True)
+@processor("mytarget-client-id", "mytarget-client-secret")
+def mytarget(**kwargs):
+    return MyTargetReader(**extract_args("mytarget_", kwargs))
+
+
+class MyTargetReader(Reader):
+
+    def __init__(
+        self,
+        client_id,
+        client_secret,
+        mail,
+        agency,
+        refresh_token,
+        **kwargs
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.mail = mail
+        self.agency = agency
+        self.agency_client_token = {'refresh_token': refresh_token}
+        self.start_date = kwargs.get('start_date')
+        self.end_date = kwargs.get('end_date')
+        self.date_format = kwargs.get('date_format')
+        self.day_range = kwargs.get('day_range')
+
+    def read(self):
+        if self.__check_date_input_validity():
```

Comment:
It would be better, if possible, to put date check in the constructor as for other readers. We don't want to create an instance of a class with bad attributes.